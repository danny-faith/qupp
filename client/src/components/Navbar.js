import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logoutUser } from './../actions/authActions';
import { clearPlaylists } from './../actions/playlistActions';
import { withRouter } from 'react-router-dom';

import logo from '../logo-v2.svg';

class Navbar extends Component {
    userRef = React.createRef();

    state = {
        isAuthenticated2: false
    }

    onLogoutClick = (e) => {
		e.preventDefault();
		this.props.clearPlaylists();
		this.props.logoutUser(this.props.history);
    }
    componentDidMount = () => {
        // console.log('Navbar componentDidMount');
        
        // if (this.state.isAuthenticated2) {
        //     window.M.Dropdown.init(this.userRef.current);
        // }
        // document.addEventListener('DOMContentLoaded', function() {
        //     var elems = document.querySelectorAll('.sidenav');
        //     var instances = M.Sidenav.init(elems, options);
        //   });
        window.M.Dropdown.init(this.userRef.current);
    }
    componentWillReceiveProps = (nextProps) => {
        // console.log(nextProps);
        
        if (nextProps.auth.isAuthenticated) {
            console.log('there are authenticated');
            // this.setState({
            //     isAuthenticated2: true
            // });
            window.M.Dropdown.init(this.userRef.current);
            
        } else {
            console.log('there are not authenticated');
            // window.M.Dropdown.init(this.userRef.current);
        }
    }
    // componentWillUpdate = () => {
    //     // console.log('componentWillUpdate');
    //     if (this.state.isAuthenticated2) {
    //         // console.log('hello?');
            
    //         window.M.Dropdown.init(this.userRef.current);
    //     } else {
    //         // console.log('me?');
            
    //     }
    // }
    render() {
        const { isAuthenticated, user } = this.props.auth;

        const authLinks = (
            <ul id="nav-mobile" className="right hide-on-med-and-down">
                <li>
                    <a href="!#" onClick={this.onLogoutClick}>Logout</a>
                </li>
                <li className="avatar">
                    <a href="!#" ref={this.userRef} data-target="dropdown1" className="dropdown-trigger">
                        <div className="avatar" style={{ backgroundImage: `url(${user.avatar})`}}>
                        </div>
                        {user.username}
                    </a>
                    <ul id="dropdown1" className="dropdown-content">
                        <li><Link to="/dashboard">Dashboard</Link></li>
                        <li><Link to="/my-account">Account</Link></li>
                        <li className="divider"></li>
                        <li><a href="!#" onClick={this.onLogoutClick}>Logout</a></li>
                    </ul>
                </li>
            </ul>
        );

        const guestLinks = (
            <ul id="nav-mobile" className="right hide-on-med-and-down">
                <li><Link to="/login">Login</Link></li>
                <li><Link to="/register">Register</Link></li>
            </ul>
        );

        return (
            <nav>
                <div className="nav-wrapper">
                <Link to="/" className="brand-logo">
                    <img width="90" alt="qupp logo" src={logo} />
                </Link>
                <a href="!#" data-target="sidenav" className="sidenav-trigger">
                    <i className="material-icons">menu</i>
                </a>
                {isAuthenticated ? authLinks : guestLinks}
                </div>
            </nav>
        )
    }
}

Navbar.propTypes = {
	clearPlaylists: PropTypes.func.isRequired,
	logoutUser: PropTypes.func.isRequired,
	auth: PropTypes.object.isRequired
}

const mapStateToProps = (state) => ({
    auth: state.auth
});

export default connect(mapStateToProps, { logoutUser, clearPlaylists })(withRouter(Navbar));