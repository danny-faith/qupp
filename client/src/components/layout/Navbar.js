import React, { useEffect } from 'react'
import { Link } from 'react-router-dom'
import { Modal } from 'react-materialize'
import Messenger from '../messenger/Messenger'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { logoutUser } from '../../actions/authActions'
import { clearPlaylists } from '../../actions/playlistActions'
import { withRouter } from 'react-router-dom'

import logo from '../../logo-v2.svg'

export function Navbar(props) {
    const dropdownTriggerRef = React.createRef()
    const messengerRef = React.createRef()
    const usersRef = React.createRef()
    const navRef = React.createRef()
    const { isAuthenticated, user } = props.auth

    useEffect(() => {
        shouldDropdownInit()
    }, [props.auth])

    const shouldDropdownInit = () => {
        if (props.auth.isAuthenticated) {
            window.M.Dropdown.init(dropdownTriggerRef.current)
        }
    }

    const onLogoutClick = (e) => {
		e.preventDefault()
		props.clearPlaylists()
		props.logoutUser(props.history)
    }
        
    const authorisedLinks = (
        <ul id="nav-mobile" className="right hide-on-med-and-down">
            <li>
                <a href="!#" onClick={onLogoutClick}>Logout</a>
            </li>
            <li>
                <Modal id="messengerUsers" className="bg-grey-darkest" header="Messenger" ref={usersRef} trigger={<a ref={messengerRef} href="!#">Messenger</a>}>
                    <Messenger />
                </Modal>
            </li>
            <li className="avatar">
                <a href="!#" ref={dropdownTriggerRef} data-target="dropdown1" className="dropdown-trigger">
                    <div className="avatar" style={{ backgroundImage: `url(${user.avatar})`}}>
                    </div>
                    {user.username}
                </a>
                <ul id="dropdown1" className="dropdown-content">
                    <li><Link to="/dashboard">Dashboard</Link></li>
                    <li><Link to="/my-account">Account</Link></li>
                    <li className="divider"></li>
                    <li><a href="!#" onClick={onLogoutClick}>Logout</a></li>
                </ul>
            </li>
        </ul>
    )

    const guestLinks = (
        <ul id="nav-mobile" className="right hide-on-med-and-down">
            <li><Link to="/login">Login</Link></li>
            <li><Link to="/register">Register</Link></li>
        </ul>
    )

    return (
        <nav ref={navRef}>
            <div className="nav-wrapper">
                <Link to="/" className="brand-logo">
                    <img width="90" alt="qupp logo" src={logo} className="ml-1" />
                </Link>
                <a href="!#" data-target="sidenav" className="sidenav-trigger">
                    <i className="material-icons">menu</i>
                </a>
                {isAuthenticated ? authorisedLinks : guestLinks}
            </div>
        </nav>
    )
}

Navbar.propTypes = {
	clearPlaylists: PropTypes.func.isRequired,
	logoutUser: PropTypes.func.isRequired,
	auth: PropTypes.object.isRequired
}

const mapStateToProps = (state) => ({
    auth: state.auth
})

export default connect(mapStateToProps, { logoutUser, clearPlaylists })(withRouter(Navbar))