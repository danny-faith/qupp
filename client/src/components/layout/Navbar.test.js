import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { Navbar } from './Navbar';

describe('Navbar shallow', () => {
  it('should render correctly', () => {
    const component = shallow(<Navbar />);
  
    expect(component).toMatchSnapshot();
  });
});

// describe('Navbar render', () => {
//   it('should render correctly', () => {
//     const component = render(<Navbar isAuthenticated="true" />);
  
//     expect(component).dive().toMatchSnapshot();
//   });
// });

// describe('Examining the syntax of Jest tests', () => {
   
//     it('sums numbers', () => {
//         expect(1 + 2).toEqual(3);
//         expect(2 + 2).toEqual(4);
//     });
// });

// import React from 'react';
// import { render, cleanup} from 'react-testing-library';
// import 'jest-dom/extend-expect';
// import Navbar from './Navbar';

// it('renders Navbar', () => {
//     const { asFragment } = render(<Navbar />).dive();
//     expect(asFragment()).toMatchSnapshot();
// });