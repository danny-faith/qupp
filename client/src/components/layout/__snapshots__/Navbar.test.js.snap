// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar shallow should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
